src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,586 @@
+
+  @classmethod
+  def implementation_version(cls):
+    return super(RscCompile, cls).implementation_version() + [('RscCompile', 7)]
+
+  @classmethod
+  def register_options(cls, register):
+    super(RscCompile, cls).register_options(register)
+
+    rsc_toolchain_version = '0.0.0-294-d7114447'
+    scalameta_toolchain_version = '4.0.0-M10'
+
+    # TODO: it would be better to have a less adhoc approach to handling
+    #       optional dependencies. See: https://github.com/pantsbuild/pants/issues/6390
+    cls.register_jvm_tool(
+      register,
+      'workaround-metacp-dependency-classpath',
+      classpath=[
+        JarDependency(org = 'org.scala-lang', name = 'scala-compiler', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-library', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-reflect', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang.modules', name = 'scala-partest_2.11', rev = '1.0.18'),
+        JarDependency(org = 'jline', name = 'jline', rev = '2.14.6'),
+        JarDependency(org = 'org.apache.commons', name = 'commons-lang3', rev = '3.3.2'),
+        JarDependency(org = 'org.apache.ant', name = 'ant', rev = '1.8.2'),
+        JarDependency(org = 'org.pegdown', name = 'pegdown', rev = '1.4.2'),
+        JarDependency(org = 'org.testng', name = 'testng', rev = '6.8.7'),
+        JarDependency(org = 'org.scalacheck', name = 'scalacheck_2.11', rev = '1.13.1'),
+        JarDependency(org = 'org.jmock', name = 'jmock-legacy', rev = '2.5.1'),
+        JarDependency(org = 'org.easymock', name = 'easymockclassextension', rev = '3.1'),
+        JarDependency(org = 'org.seleniumhq.selenium', name = 'selenium-java', rev = '2.35.0'),
+      ],
+      custom_rules=[
+        Shader.exclude_package('*', recursive=True),]
+    )
+    cls.register_jvm_tool(
+      register,
+      'rsc',
+      classpath=[
+          JarDependency(
+              org='com.twitter',
+              name='rsc_2.11',
+              rev=rsc_toolchain_version,
+          ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('rsc', recursive=True),
+      ])
+    cls.register_jvm_tool(
+      register,
+      'mjar',
+      classpath=[
+          JarDependency(
+              org='com.twitter',
+              name='mjar_2.11',
+              rev=rsc_toolchain_version,
+          ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('scala', recursive=True),
+      ])
+    cls.register_jvm_tool(
+      register,
+      'metacp',
+      classpath=[
+          JarDependency(
+            org='org.scalameta',
+            name='metacp_2.11',
+            rev=scalameta_toolchain_version,
+          ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('scala', recursive=True),
+      ])
+    cls.register_jvm_tool(
+      register,
+      'metai',
+      classpath=[
+          JarDependency(
+            org='org.scalameta',
+            name='metai_2.11',
```

Comment:
As a followup task, we need to figure out which of these tools need to match the scala version that is in use (ie, which mix with the classpath of the code under compilation, and which don't), and extract settings for them.

Reply:
I've noted the abi param in my list of things to break out into further tickets.